### ISSUE TITLE
Searing Nightmare Segmentation fault LM-SA

### ISSUE THREAD

**Issue by seanpeters86** [WarcraftPriests/sl-shadow-priest#46]
This only happens on a LightMovement sim with a raid event with 3 adds. The 3 adds likely isnt super important since that's just crossing the threshold for searing nightmare to be used in the APL. But I cannot reproduce this by using Patchwerk or HeavyMovement fight styles...which is really weird.

Profile:

```
priest="Base"
level=60
role=spell
position=back
spec=shadow
talents=0030000
spec=shadow

default_actions=1

head=,id=173218
neck=,id=158075,bonus_id=6316/4932/4933/1560,azerite_level=92
shoulder=,id=178071,bonus_id=6707,drop_level=58
back=,id=173215
chest=,id=173214
wrist=,id=173222
hands=,id=173217
waist=,id=173221
legs=,id=175431,bonus_id=6707,drop_level=58
feet=,id=173216
finger1=,id=175710,enchant_id=6165,bonus_id=6706,drop_level=57
finger2=,id=178077,enchant_id=6165,bonus_id=6707,drop_level=58
trinket1=,id=174500,gem_id=168639,bonus_id=4824/1517/4786/6514
trinket2=,id=175722,bonus_id=6707,drop_level=54
main_hand=,id=158321,enchant_id=6229,gem_id=168639,bonus_id=6536/6578/1543/4786/6476/6470/6514
off_hand=,id=172190,gem_id=168639,bonus_id=4824/1517/4786/6514/6482/6470

fight_style="LightMovement"
raid_events+=/adds,count=3,first=45,cooldown=45,duration=10,distance=5
desired_targets=1
iterations=10000
```

```
sim_signal_handler: Segmentation fault! Iteration=674 Seed=3151644859788401181 TargetHealth=422326.0
```

**Comment by seanpeters86** [comment]
This is really weird, this bug stops happening if I add this:

```
gear_haste_rating=650
```

**Comment by scamille** [comment]
Fixed by simulationcraft/simc@c2f9b1280c035181db7b4da9725f231a578a3c85
